pkg/store/prometheus_test.go

```diff
@@ -420,7 +420,11 @@ func TestPrometheusStore_LabelValues_e2e(t *testing.T) {
 	u, err := url.Parse(fmt.Sprintf("http://%s", p.Addr()))
 	testutil.Ok(t, err)
 
-	proxy, err := NewPrometheusStore(nil, nil, promclient.NewDefaultClient(), u, component.Sidecar, getExternalLabels, nil)
+	version, err := promclient.NewDefaultClient().BuildVersion(ctx, u)
```

Comment:
Let's add a test case where the `version` is a garbage value, and check how your code behaves? However, let's ask @yeya24 @bwplotka if this might occur in the first place or not 🤔

Reply:
I guess `version` could be garbage value only if `GET /api/v1/status/buildinfo` API returns some garbage value for the Prometheus version. But I think this API would be throwing some err instead of returning garbage value. Not sure though.

Reply:
Malformed version happens when a non-release prometheus is used.
For this case, I recommend using semver to get the version. If it is malformed, we just use the series way

Reply:
> If it is malformed, we just use the series way

What does `malformed version` exactly mean? Does it mean 'incorrect' version or version something like this `1.2.3-beta.1` (pre-release version)?

Reply:
You can take a look at https://semver.org/.
`1.2.3-beta.1` is valid. `Malformed version` here means the version that cannot be parsed by semver. 

For example, if you build Prometheus locally, the version would be your git branch name like `test`. `test` is not of format `x.y.z` so that's the `malformed version`.
You can also check it with https://jubianchi.github.io/semver-check/#/

Reply:
Gotta, Thanks. Let's do it.